deepchecks/vision/checks/performance/single_dataset_scalar_performance.py

```diff
@@ -0,0 +1,148 @@
+    ObjectDetectionAveragePrecision
+from deepchecks.vision.vision_data import TaskType
+
+__all__ = ['SingleDatasetScalarPerformance']
+
+
+class SingleDatasetScalarPerformance(SingleDatasetCheck):
+    """Calculate a performance metric as a scalar for a given model and a given dataset.
+
+    Parameters
+    ----------
+        metric: Metric, default: None
+        An ignite.Metric object whose score should be used. If None is given, use the default metric.
+        reduce: torch function, default: None
+        The function to reduce the scores tensor into a single scalar. For metrics that return a scalar use None
+        (default).
+    """
+
+    def __init__(self,
+                 metric: Metric = None,
+                 reduce: t.Callable = None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        self.metric = metric
+        self.reduce = reduce
+        if 'metric_name' in kwargs:
+            self.metric_name = kwargs['metric_name']
+
+        if 'reduce_name' in kwargs:
+            self.reduce_name = kwargs['reduce_name']
+
+    def initialize_run(self, context: Context, dataset_kind: DatasetKind.TRAIN):
+        """Initialize the metric for the check, and validate task type is relevant."""
+        if self.metric is None:
+            if context.train.task_type == TaskType.CLASSIFICATION:
+                self.metric = Accuracy()
+                if not hasattr(self, 'metric_name'):
+                    self.metric_name = 'accuracy'
+            elif context.train.task_type == TaskType.OBJECT_DETECTION:
+                self.metric = ObjectDetectionAveragePrecision()
+                if not hasattr(self, 'metric_name'):
+                    self.metric_name = 'object_detection_average_precision'
+                if self.reduce is None:
+                    self.reduce = torch.nanmean
+            else:
+                raise DeepchecksValueError('For task types other then classification or object detection, '
+                                           'pass a metric explicitly')
+        self.metric.reset()
+
+    def update(self, context: Context, batch: Batch, dataset_kind: DatasetKind.TRAIN):
+        """Update the metrics by passing the batch to ignite metric update method."""
+        label = batch.labels
+        prediction = batch.predictions
+        self.metric.update((prediction, label))
+
+    def compute(self, context: Context, dataset_kind: DatasetKind.TRAIN) -> CheckResult:
+        """Compute the metric result using the ignite metrics compute method and reduce to a scalar."""
+        metric_result = self.metric.compute()
+        if self.reduce is not None:
+            if isinstance(metric_result, float):
+                warnings.warn(SyntaxWarning('Metric result is already scalar, skipping reduce operation'))
+                result_value = metric_result
+            else:
+                result_value = float(self.reduce(metric_result))
```

Comment:
about same comment on the floats, do we care? maybe it's a numpy float, why not let it be a numpy float

Reply:
I'm less concerned about numpy floats and more about non float numbers such as integers

Reply:
if it's gonna be serialized, having it being a pure python float is good IMO

Reply:
well the json/wandb serialization is doing un_numpy to the value anyways, isn't critical but the user might prefer numpy

Reply:
Also kinda continuing on gabbay comment, is it possible the reduce will return a tensor? Cuz if so I think it will also fail